pageserver/src/tenant.rs

```diff
@@ -487,6 +487,33 @@ enum CreateTimelineCause {
     Delete,
 }
 
+#[derive(thiserror::Error, Debug)]
+pub(crate) enum GcError {
+    // The tenant is shutting down
+    #[error("tenant shutting down")]
+    TenantCancelled,
```

Comment:
Do we need to distinguish the two cases? Cancellation is cancellation. Also, the names are probably not accurate because at least one of the two gets invoked during pageserver shutdowns as well.

Reply:
I started out with just one Cancelled variant.  Then, I was updating the code around this, and it felt quite error prone in the cases where we want to drop out of GC when the tenant is cancelled, but not with just one timeline is cancelled.

TLDR: we could make do with just one Cancelled, but I think it would make the code less obvious in the places we handle it.

Reply:
Hmmm I see that explanation makes sense, approving